### PR TITLE
GEM-115 | Get user expenses GET /internal/expenses/groups/{group_id}/users/{user_id}

### PR DESCRIPTION
**Pull request by pwgacek**

### CODE REVIEW

File: src/main/kotlin/pl/edu/agh/gem/internal/mapper/CreditorUserExpenseMapper.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package pl.edu.agh.gem.internal.mapper
+
+import pl.edu.agh.gem.internal.model.expense.Expense
+import pl.edu.agh.gem.internal.model.expense.ExpenseStatus.ACCEPTED
+import pl.edu.agh.gem.internal.model.expense.UserExpense
+import java.math.BigDecimal
+
+class CreditorUserExpenseMapper {
+    fun mapToUserExpense(creditorId: String, expense: Expense): UserExpense? {
```
Comment (lawos98) [inline src/main/kotlin/pl/edu/agh/gem/internal/mapper/CreditorUserExpenseMapper.kt:9]: maybe more like 
```kotlin
fun mapToUserExpense(creditorId: String, expense: Expense): UserExpense? =
  when {
    expense.status != ACCEPTED -> null
    creditorId != expense.creatorId -> null
    else -> UserExpense(
      value = getCostAsExpenseCreator(expense),
      currency = expense.targetCurrency ?: expense.baseCurrency,
      exchangeRate = expense.exchangeRate
    )
  }
```

File: src/main/kotlin/pl/edu/agh/gem/internal/mapper/DebtorUserExpenseMapper.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package pl.edu.agh.gem.internal.mapper
+
+import pl.edu.agh.gem.internal.model.expense.Expense
+import pl.edu.agh.gem.internal.model.expense.ExpenseStatus.ACCEPTED
+import pl.edu.agh.gem.internal.model.expense.UserExpense
+
+class DebtorUserExpenseMapper {
+    fun mapToUserExpense(debtorId: String, expense: Expense): UserExpense? {
```
Comment (lawos98) [inline src/main/kotlin/pl/edu/agh/gem/internal/mapper/DebtorUserExpenseMapper.kt:8]: maybe more like 

```kotlin
fun mapToUserExpense(debtorId: String, expense: Expense): UserExpense? =
val cost = expense.expenseParticipants.find { it.participantId == debtorId }?.participantCost
  when {
    expense.status != ACCEPTED -> null
    expense.creatorId == debtorId -> null
    cost == null -> null
    else -> UserExpense(
      value = cost.negate(),
      currency = expense.targetCurrency ?: expense.baseCurrency,
      exchangeRate = expense.exchangeRate
    )
  }
```

File: src/main/kotlin/pl/edu/agh/gem/internal/mapper/CreditorUserExpenseMapper.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        if (creditorId != expense.creatorId) return null
+        return UserExpense(
+            value = getCostAsExpenseCreator(expense),
+            currency = expense.targetCurrency ?: expense.baseCurrency,
+            exchangeRate = expense.exchangeRate,
+        )
+    }
+
+    private fun getCostAsExpenseCreator(expense: Expense): BigDecimal {
+        return expense.expenseParticipants
```
Comment (lawos98) [inline src/main/kotlin/pl/edu/agh/gem/internal/mapper/CreditorUserExpenseMapper.kt:20]: maybe more like
```kotlin
private fun getCostAsExpenseCreator(expense: Expense): BigDecimal =
  expense.expenseParticipants.filterNot { it.participantId == expense.creatorId }
    .sumBy { it.participantCost }
```

**Comment by pwgacek** [comment]
also added userId to the response

**Review by lawos98 — APPROVED** [review]

**Pull request merged**